tests/components/wemo/test_init.py

```diff
@@ -0,0 +1,67 @@
+"""Tests for the wemo component."""
+from homeassistant.components.wemo import CONF_DISCOVERY, CONF_STATIC
+from homeassistant.components.wemo.const import DOMAIN
+from homeassistant.setup import async_setup_component
+
+from .conftest import MOCK_HOST, MOCK_PORT
+
+
+async def test_config_no_config(hass):
+    """Component setup succeeds when there are no config entry for the domain."""
+    assert await async_setup_component(hass, DOMAIN, {})
+
+
+async def test_config_no_static(hass):
+    """Component setup succeeds when there are no static config entries."""
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_DISCOVERY: False}})
+
+
+async def test_static_config_with_port(hass, pywemo_device):
+    """Static device with host and port is added and removed."""
```

Comment:
I am trying to learn more about testing - do these tests implicitly test removal?  The adding seems explicit but I don't see device removal.  Thanks!

Reply:
These tests don't test removal. That's because the component doesn't currently implement `async_unload_entry` AFAICT that method needs to be implemented at the component level before removal is possible. So, for these tests the test fixtures (ex `hass`) are destroyed after each test run, and that's what is currently removing the entities.

If unloading was supported (which I'm hoping to add support for soon), I believe this is the appropriate way to unload:
```
    # Unload the entry to trigger any cleanups.
    assert await hass.config_entries.async_unload(entity_entries[0].config_entry_id)
    await hass.async_block_till_done()
```